## Lesson 4 Homework

In [1]:
import pandas as pd

In [2]:
chipotle_data = pd.read_table('../DS-SEA-06/data/chipotle.tsv')

chipotle_data.head()

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98

In [3]:
chipotle_data.tail()

order_id  quantity           item_name  \
4617      1833         1       Steak Burrito   
4618      1833         1       Steak Burrito   
4619      1834         1  Chicken Salad Bowl   
4620      1834         1  Chicken Salad Bowl   
4621      1834         1  Chicken Salad Bowl   

                                     choice_description item_price  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75

Each row is the number of units of an item in one order and the total price of the number of those units ordered. An order consists of one or more rows.

In [6]:
orders = chipotle_data.groupby('order_id')

In [12]:
orders.order_id.nunique()

order_id
1       1
2       1
3       1
4       1
5       1
6       1
7       1
8       1
9       1
10      1
11      1
12      1
13      1
14      1
15      1
16      1
17      1
18      1
19      1
20      1
21      1
22      1
23      1
24      1
25      1
26      1
27      1
28      1
29      1
30      1
       ..
1805    1
1806    1
1807    1
1808    1
1809    1
1810    1
1811    1
1812    1
1813    1
1814    1
1815    1
1816    1
1817    1
1818    1
1819    1
1820    1
1821    1
1822    1
1823    1
1824    1
1825    1
1826    1
1827    1
1828    1
1829    1
1830    1
1831    1
1832    1
1833    1
1834    1
Name: order_id, dtype: int64

There are **1834 orders** in this file.

There are **4621 lines** in this file.

In [21]:
items = chipotle_data.groupby('item_name')

In [41]:
len(items.groups['Chicken Burrito'])

553

In [42]:
len(items.groups['Steak Burrito'])

368

The *Chicken Burrito* is **more** popular than the *Steak Burrito*

In [46]:
items.get_group('Chicken Burrito').tail()

order_id  quantity        item_name  \
4556      1811         1  Chicken Burrito   
4558      1812         1  Chicken Burrito   
4559      1812         1  Chicken Burrito   
4566      1816         1  Chicken Burrito   
4569      1817         1  Chicken Burrito   

                                     choice_description item_price  
4556  [Tomatillo Green Chili Salsa, [Fajita Vegetabl...    $11.25   
4558  [Fresh Tomato Salsa, [Rice, Black Beans, Chees...     $8.75   
4559  [Tomatillo Red Chili Salsa, [Rice, Cheese, Gua...    $11.25   
4566  [Roasted Chili Corn Salsa, [Pinto Beans, Chees...     $8.75   
4569    [Fresh Tomato Salsa, [Fajita Vegetables, Rice]]     $8.75

In [48]:
import os
import subprocess

aloki@sango MINGW64 ~/GA/DS-SEA-06 (master)
$ find|grep sv$
./data/airlines.csv
./data/Airline_on_time_west_coast.csv
./data/bank-additional.csv
./data/bikeshare.csv
./data/chipotle.tsv
./data/citibike_feb2014.csv
./data/drinks.csv
./data/drones.csv
./data/hitters.csv
./data/icecream.csv
./data/imdb_1000.csv
./data/mtcars.csv
./data/NBA_players_2015.csv
./data/ozone.csv
./data/pronto_cycle_share/2015_station_data.csv
./data/pronto_cycle_share/2015_trip_data.csv
./data/pronto_cycle_share/2015_weather_data.csv
./data/rossmann.csv
./data/rt_critics.csv
./data/sms.tsv
./data/stores.csv
./data/syria.csv
./data/time_series_train.csv
./data/titanic.csv
./data/ufo.csv
./data/vehicles_test.csv
./data/vehicles_train.csv
./data/yelp.csv
